lib/auth/integration/integrationv1/service.go

```diff
@@ -248,12 +248,8 @@ func (s *Service) CreateIntegration(ctx context.Context, req *integrationpb.Crea
 			return nil, trace.Wrap(err)
 		}
 	case types.IntegrationSubKindAWSOIDC, types.IntegrationSubKindAWSRolesAnywhere:
-		// AWS OIDC and Roles Anywhere Integrations can be used as source of credentials to access AWS Web/CLI.
-		// For OIDC, this creates a new AppServer whose endpoint is <integrationName>.<proxyURL>, which can fail if integrationName is not a valid DNS Label.
-		// For Roles Anywhere, this creates a AppServers for each Roles Anywhere Profile whose endpoint is <profileName>-<integrationName>.<proxyURL>, which can fail if integrationName is not a valid DNS Label.
-		// Instead of failing when the integration is already created, it fails at creation time.
-		if errs := validation.IsDNS1035Label(req.GetIntegration().GetName()); len(errs) > 0 {
-			return nil, trace.BadParameter("integration name %q must be a lower case valid DNS subdomain so that it can be used to allow Web/CLI access", req.GetIntegration().GetName())
+		if err := awscommon.ValidIntegratioName(req.Integration.GetName()); err != nil {
```

Comment:
Like discussed in Slack, I am not sure if this was a wise idea to force dns name validation outside the usage of aws role anywhere/cli access. The ux issue this PR is trying to solve is exactly because it forced changed the resource naming compatibility without updating all of the current usage.

Reply:
> This PR doesn't change the AWS OIDC integration flows.
```
case types.IntegrationSubKindAWSOIDC, types.IntegrationSubKindAWSRolesAnywhere:
		if err := awscommon.ValidIntegratioName(req.Integration.GetName()); err != nil {
```
Sorry I got confused since the code changes follows the OIDC case path, I thought to mention it here. 

But please be careful to not to break naming compatibility in future. It unnecessarily consumes engineering resources now that other integrations that depends on OIDC integration needs to be updated too.